samples/bluetooth/peripheral_lbs/src/main.c

```diff
@@ -221,11 +215,17 @@ static void bt_ready(int err)
 	k_sem_give(&ble_init_ok);
 }
 
+
+static void send_button_state(struct k_work *item)
+{
+	lbs_send_button_state(!button_state);
+}
+
 static void button_pressed(struct device *button_port, struct gpio_callback *cb,
 		    u32_t pins)
 {
-	button_state = button_state ^ 0x01;
-	lbs_send_button_state(button_state);
+	gpio_pin_read(button_port, USER_BUTTON, (u32_t *)&button_state);
```

Comment:
I recommend moving the `gpio_pin_read` call to the `send_button_state` function.
It should add some filtering of button state changes and removes the race between set and get on button state.